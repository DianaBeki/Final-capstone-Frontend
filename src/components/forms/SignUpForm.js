import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { signUpUser } from '../../redux/user/userSlice';

const SignUpForm = () => {
    const dispatch = useDispatch();
    const navigate = useNavigate();
    const [name, setName] = useState('');
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
  
    const handleSignUp = async (e) => {
        e.preventDefault();
    
        try {
          await dispatch(signUpUser({ name, email, password }));
          navigate.push('/doctors');
        } catch (error) {
          console.error('Sign Up Error:', error);
        }
      };

      return (
        <div className="flex items-center justify-center h-screen">
          <form className="bg-white p-8 shadow-md rounded-md w-96" onSubmit={handleSignUp}>
            <h2 className="text-2xl font-semibold mb-6">Sign Up</h2>
            <div className="mb-4">