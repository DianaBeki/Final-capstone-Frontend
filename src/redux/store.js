import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
