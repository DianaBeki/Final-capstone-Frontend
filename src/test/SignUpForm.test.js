import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import SignUpForm from '../components/forms/SignUpForm';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    <Router>
      {component}
    </Router>
  </Provider>,
);

describe('SignUpForm Component', () => {
  it('renders SignUpForm component', () => {
    const { container } = render(<SignUpForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
