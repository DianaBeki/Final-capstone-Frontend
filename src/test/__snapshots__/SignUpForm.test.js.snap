// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm Component renders SignUpForm component 1`] = `
<div
  class="flex items-center justify-center h-screen"
>
  <form
    class="bg-white p-8 shadow-md rounded-md w-96"
  >
    <h2
      class="text-2xl font-semibold mb-6"
    >
      Sign Up
    </h2>
    <div
      class="mb-4"
    >
      <input
        class="mt-1 p-2 w-full border rounded-md"
        id="name"
        name="name"
        placeholder="Enter your name"
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-4"
    >
      <input
        class="mt-1 p-2 w-full border rounded-md"
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      class="mb-4"
    >
      <input
        class="mt-1 p-2 w-full border rounded-md"
        id="password"
        name="password"
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <button
      class="bg-blue-500 text-white px-4 py-2 rounded-md hover:bg-blue-600"
      type="submit"
    >
      Sign Up
    </button>
    <p
      class="mt-2"
    >
      have an account?
       
      <a
        href="/login"
      >
        Click here to log in
      </a>
    </p>
  </form>
</div>
`;
